tests/path.py

```diff
@@ -145,7 +145,7 @@ def text(self, encoding='utf-8', **kwargs):
         mode = 'rU' if PY2 else 'r'
         with open(self, mode=mode, encoding=encoding, **kwargs) as f:
             text = f.read()
-        contents = repr_as(text, '<%s contents>' % self.basename())
+        contents = repr_as(text, '<%s contents: %r>' % (self.basename(), text))
```

Comment:
AFAIK, `repr_as` is introduced to suppress contents on 1) list of testcases and 2) failure report.

 1) is like this:
https://travis-ci.org/sphinx-doc/sphinx/jobs/188727836#L517


I agree the second purpose make hard to debug on failure.
But this change messes up the merit of `repr_as`.

If pytest shows human-readable list of testcases, `repr_as` is no longer needed. So we should remove it.